### PR TITLE
Plain component properties now can be applied from CSS

### PR DESCRIPTION
**Pull request by enchev**
Implemented #1660

### CODE REVIEW

**Comment by hshristov** [comment]
:+1:

File: ui/styling/css-selector.ts [hunk 1]
```diff
@@ -57,11 +57,15 @@ export class CssSelector {
 
     public apply(view: view.View) {
         this.eachSetter((property, value) => {
-            try {
-                view.style._setValue(property, value, observable.ValueSource.Css);
-            }
-            catch (ex) {
-                trace.write("Error setting property: " + property.name + " view: " + view + " value: " + value + " " + ex, trace.categories.Style, trace.messageType.error);
+            if (property in view) {
```
Comment (vakrilov) [inline ui/styling/css-selector.ts]: This will probably not handle `specialProperties`. For example `row`, `column` properties from GridLayout

**Comment by vakrilov** [comment]
Perhaps verify that special properties work too.
Otherwise :+1:

**Comment by atanasovg** [comment]
Guys, in my [comment here](https://github.com/NativeScript/NativeScript/issues/1660#issuecomment-191207755) I meant that we may enable component properties by following our current path - by creating new `style-property` within the `style` module - like `color`  for example: https://github.com/NativeScript/NativeScript/blob/master/ui/styling/style.ts#L937. Sorry for not being clear enough. 

This new implementation treats applying a property through CSS with a `local` value priority, while the implementation of the [style property](https://github.com/NativeScript/NativeScript/blob/master/ui/styling/style-property.ts#L85-L109) adds new modifiers like `visualState` and `css` to the `effectiveValue` calculation.

Besides, I remember us talking some time ago on this very topic and we decided against enabling each and every property applicable through CSS mainly due to some properties simply not fitting within the styling paradigm - like `text`, `value`, `items`, etc. Can we reiterate this and evaluate the pros and cons?

Ping @enchev, @hshristov, @vakrilov

**Comment by enchev** [comment]
We've talked with @hshristov and @vakrilov about the inheritance and we've concluded that such non style properties are non inheritable and can be set as local values. For example `StackLayout` `orientation` cannot and should not be inherited. 

Personally I support this since it will give developers a lot more power.

**Comment by NathanaelA** [comment]
@enchev - Point of confusion at least for me.   Are you saying **StackLayout.orientation** is not inheritable as in the property doesn't get inherited by its children; or are you saying that it isn't going to be settable at all via the css mechanism?

**Comment by enchev** [comment]
@NathanaelA Not inheritable in the CSS world. No CSS cascading as well.

**Pull request merged**